In [10]:
import numpy as np
import pandas as pd
import sys
import matplotlib
import matplotlib.pyplot as plt

df = pd.read_csv('C:/Users/User/Downloads/data1.csv')

print(df.head()) 

   Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1
1        60    117       145     479.0
2        60    103       135     340.0
3        45    105       175     282.4
4        45    117       148     406.0


**Percentiles**

In [4]:
# Percentiles
Maxpulse= df["Maxpulse"]
percentile20 = np.percentile(Maxpulse, 20)

print(percentile20)

120.6


The 20% percentile of Max_Pulse is 120,6. This means that 20% of all the training sessions have a Max_Pulse of 120,6 or lower.

If we flip the statement, it means that 80% of all of the training sessions have an average pulse of 120.6 beats per minute or higher

**Standard Deviation**

In [5]:
std = np.std(df)
print(std)

Duration     42.174616
Pulse        14.478355
Maxpulse     16.401692
Calories    265.566544
dtype: float64


Standard deviation is a number that describes how spread out the observations are. If a standard deviation is low, it means that most numbers are close to average value. A high standard deviation means that values have a wider range

**Variance**

In [14]:
var = np.var(df)
print(var)

Duration     1778.698225
Pulse         209.622772
Maxpulse      269.015511
Calories    70525.589051
dtype: float64


Variance is another number that indicates the spread of values.

In fact, if you take square root of the variance, you get the standard deviation. And exactly opposite, if you multiply standard deviation by itself, you get variance.

**Correlation**

In [17]:
matplotlib.use('Agg')
correlation = df['Pulse'].corr(df['Maxpulse'])
print(correlation)

0.7833337288319593


A statistical measure that describes the extent to which two variables change together. It ranges from -1 to 1, where 1 indicates a perfect positive correlation, -1 indicates a perfect negative correlation, and 0 indicates no correlation.

**Correlation Matrix**

In [18]:
Corr_Matrix = round(df.corr(),2)
print(Corr_Matrix)

          Duration  Pulse  Maxpulse  Calories
Duration      1.00  -0.15      0.01      0.92
Pulse        -0.15   1.00      0.78      0.02
Maxpulse      0.01   0.78      1.00      0.20
Calories      0.92   0.02      0.20      1.00


A table showing correlation coefficients between variables. It's useful for understanding relationships between multiple variables in a dataset.

**Correlation vs. Causality**:
Correlation does not imply causality. Just because two variables are correlated does not mean that one causes the other. Correlation simply indicates that there is a relationship between the variables.